rpc/handlers.go

```diff
@@ -267,6 +268,11 @@ func adaptTransaction(t core.Transaction) *Transaction {
 	default:
 		panic("not a transaction")
 	}
+
+	if txn.Version.IsZero() && txn.Type != TxnL1Handler {
+		txn.Nonce = nil
+	}
+	return txn
```

Comment:
@wojciechos I did checked from the spec, but can you double check that this is correct.

Reply:
Yeah, seems to be correct.